### PR TITLE
feat(theme-docs): allow customising default branch

### PR DESCRIPTION
**Pull request by danielroe**
## Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Not everyone will be preparing PRs against `dev`. It may be directly against `main` or against another branch. This PR allows customising this in settings (or automatically fetching the branch from GitHub). It defaults to `dev`, although this will be a breaking change in the (unlikely) case that users *prefer* a diff against `dev` even though they actually have a different default branch.

## Checklist:
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

**Comment by danielroe** [comment]
Note this still hard codes PRs against `docs/content` prefix, which we should be able to detect and auto-update at build time (though I think that is for another day!)

File: packages/theme-docs/src/store/index.js [hunk 1]
```diff
@@ -86,6 +90,28 @@ export const actions = {
...
+      const data = await fetch(`https://api.github.com/repos/${state.settings.github}`, options).then((res) => {
+        if (!res.ok) {
+          throw new Error(res.statusText)
+        }
+        return res
+      }).then(res => res.json())
+      defaultBranch = data.default_branch
+    } catch (e) {}
+
+    commit('SET_DEFAULT_BRANCH', defaultBranch || 'dev')
```
Comment (benjamincanac) [inline packages/theme-docs/src/store/index.js:113]: @danielroe Maybe this should be set to `master` by default?

Comment (danielroe) [reply packages/theme-docs/src/store/index.js:113]: That would be less opinionated, although at some point that will switch to main, and so we could just start with that?

https://github.com/github/renaming

Comment (danielroe) [inline packages/theme-docs/src/store/index.js:113]:
```suggestion
    commit('SET_DEFAULT_BRANCH', defaultBranch || 'main')
```

Comment (danielroe) [reply packages/theme-docs/src/store/index.js]: let me know if you prefer `master` and I can adjust PR

File: docs/content/en/themes-docs.md [hunk 2]
```diff
@@ -210,6 +210,9 @@ You can create a `content/settings.json` file to configure the theme:
 - `github`
   - Type: `String`
   - *The GitHub repository of your project `${org}/${name}` to display the last version, the releases page, the link at the top and the `Edit this page on GitHub link` on each page*
+- `defaultBranch`
```
Comment (benjamincanac) [inline docs/content/en/themes-docs.md:213]: @danielroe Could you add the default value to `main` here?

File: packages/theme-docs/src/store/index.js (hunk 1 above)
Comment (benjamincanac) [reply packages/theme-docs/src/store/index.js]: `main` is fine :)

File: docs/content/en/themes-docs.md [hunk 3]
```diff
@@ -210,6 +210,9 @@ You can create a `content/settings.json` file to configure the theme:
...
+  - Type: `String`
+  - *The default branch for the GitHub repository of your project, used in the `Edit this page on GitHub link` on each page.*
```
Comment (danielroe) [inline docs/content/en/themes-docs.md:215]:
```suggestion
  - *The default branch for the GitHub repository of your project, used in the `Edit this page on GitHub link` on each page (defaults to `main` if it cannot be detected)*
```

File: docs/content/en/themes-docs.md (hunk 2 above)
Comment (danielroe) [reply docs/content/en/themes-docs.md:213]: :+1: Done.

**Comment by benjamincanac** [comment]
Thanks @danielroe!

**Pull request merged**